packages/big-design/src/components/Dropdown/Dropdown.tsx

```diff
@@ -111,17 +111,19 @@ export class Dropdown extends React.PureComponent<DropdownProps, DropdownState>
     );
   }
 
-  private renderIcon(option: DropdownItem | DropdownLinkItem, isHighlighted: boolean) {
+  private renderIcon(item: Item<T> | LinkItem<T>, isHighlighted: boolean) {
+    const getColor = () => {
+      if (item.disabled) {
+        return 'secondary40';
+      } else {
+        return isHighlighted ? (item.actionType === 'destructive' ? 'danger50' : 'primary') : 'secondary60';
+      }
+    };
```

Comment:
🍹

```js
const getColor = () => {
  if (item.disabled) {
    return 'secondary40';
  }

  if (!isHighlighted) {
    return 'secondary60';
  }

  return item.actionType === 'destructive' ? 'danger50' : 'primary';
}
```

Reply:
🍹 I'd also move `getColor` to a method in the `Dropdown` class.

Reply:
The reason why I didn't move it to it's own method in the class is because it needs access to `item` and `isHighlighted`. Would you still do it that way and just pass the params again?